_includes/markdown/PHP.md

```diff
@@ -350,17 +350,41 @@ Using a common set of design patterns while working with PHP code is the easiest
 
 #### Namespacing
 
-All functional code should be properly namespaced. We do this to logically organize our code and to prevent collisions in the global namespace. Generally, this means using a PHP ```namespace``` identifier at the top of included files:
+We properly namespace all PHP code outside of theme templates. This means any PHP file that isn't part of the [WordPress Template Hierarchy](https://developer.wordpress.org/themes/basics/template-hierarchy/) should be organized within a namespace or _pseudo_ namespace so its contents don't conflict with other, similarly-named classes and functions ("namespace collisions").
+
+Generally, this means using a PHP ```namespace``` identifier at the top of included files:
 
 ```php
 <?php
-namespace tenup\Utilities\API;
+namespace TenUp\Tyrell-Corporation\Nexus-6;
 
 function do_something() {
   // ...
 }
 ```
 
+A namespace identifier consists of a _top-level_ namespace or "Vendor Name", which is usually ```TenUp``` for our projects. We follow the top-level name with a project name, usually a client's name, and then with the name of a particular site or high-level project we're building. Themes and plugins written for the same client should share a namespace. If we have two identically-name classes or functions in the same namespace, that's a sign there might be duplicate code we can get rid of.
+  
+When we manage multiple sites for a single client and create a _shared plugin_ for common functionality, it should have its own namespace named "Common" (example: TenUp\Tyrell-Corporation\Common").
```

Comment:
Personally, I don't think common plugins should be treated any different from themes. For example, if `Tyrell_Corporation` has a Twitter plugin named "twitter", the namespace should be `TenUp\Tyrell_Corporation\Twitter` because that is the "code space" proper to that set of code. If this is an internal project, it seems to me we could skip the "client" part and just do "vendor\project" (e.g. `TenUp\ElasticPress`).